src/django_components/dependencies.py

```diff
@@ -266,21 +389,30 @@ def postprocess_component_html(
 TContent = TypeVar("TContent", bound=Union[bytes, str])
 
 
-CSS_DEPENDENCY_PLACEHOLDER = '<link name="CSS_PLACEHOLDER">'
-JS_DEPENDENCY_PLACEHOLDER = '<script name="JS_PLACEHOLDER"></script>'
-
-CSS_PLACEHOLDER_BYTES = bytes(CSS_DEPENDENCY_PLACEHOLDER, encoding="utf-8")
-JS_PLACEHOLDER_BYTES = bytes(JS_DEPENDENCY_PLACEHOLDER, encoding="utf-8")
+CSS_PLACEHOLDER_NAME = "CSS_PLACEHOLDER"
+CSS_PLACEHOLDER_NAME_B = CSS_PLACEHOLDER_NAME.encode()
+JS_PLACEHOLDER_NAME = "JS_PLACEHOLDER"
+JS_PLACEHOLDER_NAME_B = JS_PLACEHOLDER_NAME.encode()
 
+CSS_DEPENDENCY_PLACEHOLDER = f'<link name="{CSS_PLACEHOLDER_NAME}"/>'
+JS_DEPENDENCY_PLACEHOLDER = f'<script name="{JS_PLACEHOLDER_NAME}"></script>'
 COMPONENT_DEPS_COMMENT = "<!-- _RENDERED {data} -->"
-# E.g. `<!-- _RENDERED table,123 -->`
-COMPONENT_COMMENT_REGEX = re.compile(rb"<!-- _RENDERED (?P<data>[\w\-,/]+?) -->")
-# E.g. `table,123`
-SCRIPT_NAME_REGEX = re.compile(rb"^(?P<comp_cls_hash>[\w\-\./]+?),(?P<id>[\w]+?)$")
+
+# E.g. `<!-- _RENDERED table,123,a92ef298,bd002c3 -->`
+COMPONENT_COMMENT_REGEX = re.compile(rb"<!--\s+_RENDERED\s+(?P<data>[\w\-,/]+?)\s+-->")
+# E.g. `table,123,a92ef298,bd002c3`
+SCRIPT_NAME_REGEX = re.compile(
+    rb"^(?P<comp_cls_hash>[\w\-\./]+?),(?P<id>[\w]+?),(?P<js>[0-9a-f]*?),(?P<css>[0-9a-f]*?)$"
+)
+# E.g. `data-djc-id-a1b2c3`
+MAYBE_COMP_ID = r"(?: data-djc-id-\w{6})?"
+# E.g. `data-djc-css-99914b`
+MAYBE_COMP_CSS_ID = r"(?: data-djc-css-\w{6})?"
+
 PLACEHOLDER_REGEX = re.compile(
     r"{css_placeholder}|{js_placeholder}".format(
-        css_placeholder=CSS_DEPENDENCY_PLACEHOLDER,
-        js_placeholder=JS_DEPENDENCY_PLACEHOLDER,
+        css_placeholder=f'<link{MAYBE_COMP_CSS_ID}{MAYBE_COMP_ID} name="{CSS_PLACEHOLDER_NAME}"/>',
```

Comment:
One peculiar thing that now happens is that we add the `data-djc-id-` attributes to the root elements. But if we define the `{% component_js/css_dependencies %}` template tags at the root of component's template, e.g.

```py
class Table(Component):
    template = """
    {% component_js_dependencies %}
    <div> Hello </div>
    """
```

Then since `{% component_js_dependencies %}` simply renders a dummy `<script>` tag, then this script tag will be assigned with `data-djc-id-` too.

And so in the regex for finding and removing these placeholder tags, we have to account for that the tags may have that attribute.